File: contrib/pyre/tests/libpyre/journal/firewall.cc
// -*- coding: utf-8 -*-
//
// michael a.g. aïvázis
// orthologue
// (c) 1998-2019 all rights reserved
//


// for the build system
#include <portinfo>

// packages
#include <cassert>
// access to the journal header file
#include <pyre/journal.h>

// main program
int main() {

    // instantiate a firewall channel
    pyre::journal::firewall_t firewall("pyre.journal.test");
    // check that it is inactive, by default
    assert(firewall.isActive() == true);

    // activate it
    firewall.deactivate();
    // and check
    assert(firewall.isActive() == false);

    // now, instantiate again using the same channel name
    pyre::journal::firewall_t again("pyre.journal.test");
    // check that it is active
    assert(again.isActive() == false);

    // all done
    return 0;
}

// end of file

File: extern/pyre/tests/journal/firewall.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis
# orthologue
# (c) 1998-2019 all rights reserved
#


"""
Verify that firewall channels by the same name share a common state
"""


def test():
    # access the package
    import journal
    # build a firewall channel
    firewall = journal.firewall("journal.test1")
    # verify that it is on by default
    assert firewall.active == True
    # disable it
    firewall.active = False

    # access the same channel through another object
    clone = journal.firewall("journal.test1")
    # verify that it is now off
    assert clone.active == False

    # build a firewall channel with a different name
    another = journal.firewall("journal.test2")
    # verify that it is on by default, to make sure that there is no crosstalk between channels
    assert another.active == True

    # all done
    return


# main
if __name__ == "__main__":
    test()


# end of file
